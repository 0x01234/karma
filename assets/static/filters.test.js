test("filters addBadge()", () => {
    // mock data attr with default filter
    //document.body.innerHTML = "<div id='filter' data-default-filter='foo=bar'></div>";
    window.jQuery = require("jquery");
    const filters = require("./filters");
    filters.addBadge("foo=bar");
});
