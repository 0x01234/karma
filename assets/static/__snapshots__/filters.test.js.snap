// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appended filtes should be present in history 1`] = `
"<li role=\\"separator\\" class=\\"divider\\"></li>
  
    
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-home\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>"
`;

exports[`appended filtes should be present in history 2`] = `
"<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>

    
  


  <li role=\\"separator\\" class=\\"divider\\"></li>
  
    
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-home\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>"
`;

exports[`appended filtes should be present in history 3`] = `
"<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default,bar
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
      <span class=\\"label-list label label-info\\">
        bar
      </span>
    
  </a>
</li>

    
  
    
      
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>

    
  


  <li role=\\"separator\\" class=\\"divider\\"></li>
  
    
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-home\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>"
`;

exports[`appended filtes should be present in history 4`] = `
"<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default,bar,@state=active
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
      <span class=\\"label-list label label-info\\">
        bar
      </span>
    
      <span class=\\"label-list label label-info\\">
        @state=active
      </span>
    
  </a>
</li>

    
  
    
      
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default,bar
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
      <span class=\\"label-list label label-info\\">
        bar
      </span>
    
  </a>
</li>

    
  
    
      
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>

    
  


  <li role=\\"separator\\" class=\\"divider\\"></li>
  
    
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-home\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>"
`;

exports[`appended filtes should be present in history 5`] = `
"<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default,bar,@state=active
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
      <span class=\\"label-list label label-info\\">
        bar
      </span>
    
      <span class=\\"label-list label label-info\\">
        @state=active
      </span>
    
  </a>
</li>

    
  
    
      
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default,bar
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
      <span class=\\"label-list label label-info\\">
        bar
      </span>
    
  </a>
</li>

    
  
    
      
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>

    
  


  <li role=\\"separator\\" class=\\"divider\\"></li>
  
    
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-home\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>"
`;

exports[`appended filtes should be present in history 6`] = `
"<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      @state=active
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        @state=active
      </span>
    
  </a>
</li>

    
  
    
      
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default,bar,@state=active
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
      <span class=\\"label-list label label-info\\">
        bar
      </span>
    
      <span class=\\"label-list label label-info\\">
        @state=active
      </span>
    
  </a>
</li>

    
  
    
      
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default,bar
    </span>
    <i class=\\"fa fa-search\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
      <span class=\\"label-list label label-info\\">
        bar
      </span>
    
  </a>
</li>

    
  


  <li role=\\"separator\\" class=\\"divider\\"></li>
  
    
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-home\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>"
`;

exports[`default filter should be in history after setting filter to foo 1`] = `
"<li role=\\"separator\\" class=\\"divider\\"></li>
  
    
<li class=\\"history-menu\\">
  <a class=\\"cursor-pointer history-menu-item\\">
    <span class=\\"rawFilter hidden\\">
      default
    </span>
    <i class=\\"fa fa-home\\"></i>
    
      <span class=\\"label-list label label-info\\">
        default
      </span>
    
  </a>
</li>"
`;
